### PR TITLE
Draft of ros_can and static inspection A

### PR DESCRIPTION
**Pull request by daniel-bara**
We will have to figure out types (either use `eufs_msgs` or make our own)

We will have to test this on the car

### CODE REVIEW

File: ros/fsai_api/src/ros_can.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        // Loop timing
+        usleep(timing_us);
+    }
+
+    return nullptr;
+}
+
+int main(int argc, char** argv) {
+    ros::init(argc, argv, "ackermann_can");
```
Comment (ugsfume) [inline ros/fsai_api/src/ros_can.cpp:74]:
```suggestion
    ros::init(argc, argv, "ros_can");
```

File: ros/fsai_api/src/ros_can.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    ros::NodeHandle nh;
+
+    if (argc < 2) {
+        ROS_ERROR("Usage: rosrun fsai_api ackermann_can <can_interface>");
```
Comment (ugsfume) [inline ros/fsai_api/src/ros_can.cpp:78]:
```suggestion
        ROS_ERROR("Usage: rosrun fsai_api ros_can <can_interface>");
```

File: ros/fsai_api/src/static inspection A.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+        // Loop timing
+        usleep(timing_us);
+    }
+
+    return nullptr;
+}
+
+int main(int argc, char** argv) {
+    ros::init(argc, argv, "ackermann_can");
```
Comment (ugsfume) [inline ros/fsai_api/src/static inspection A.cpp:133]:
```suggestion
    ros::init(argc, argv, "static_inspection_A");
```

File: ros/fsai_api/src/static inspection A.cpp [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    ros::NodeHandle nh;
+
+    if (argc < 2) {
+        ROS_ERROR("Usage: rosrun fsai_api ackermann_can <can_interface>");
```
Comment (ugsfume) [inline ros/fsai_api/src/static inspection A.cpp:137]:
```suggestion
        ROS_ERROR("Usage: rosrun fsai_api static_inspection_A <can_interface>");
```

**Review by ugsfume — COMMENTED** [review]
Could you rename static inspection A.cpp to static_inspect_A.cpp

File: ros/fsai_api/src/ros_can.cpp (hunk 2 above)
Comment (daniel-bara) [reply ros/fsai_api/src/ros_can.cpp:78]: Suggestion applied

File: ros/fsai_api/src/static inspection A.cpp (hunk 4 above)
Comment (daniel-bara) [reply ros/fsai_api/src/static inspection A.cpp:137]: Suggestion applied

**Comment by ugsfume** [comment]
I'd like to keep ackermann_ros.cpp and handshake.cpp here.

While we can hardcode static_inspection_A routine into a single node. I think it's best to have a general translator node that listens to /ackermann_cmd, and translate into CAN APIs. So the routines can be programmed in python.

**Pull request merged**